File: services/flashcard/src/main/kotlin/si/um/kotnik/application/dtos/AttemptDto.kt
package si.um.kotnik.application.dtos

import java.time.LocalDateTime
import java.time.ZoneOffset

data class AttemptDto (
    val userEmail: String,
    val startedAt: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC),
    val endedAt: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC),
    val hasPassed: Boolean = false,
    val cardSetId: String
)

File: bff/web/WebBFF/Application/Dtos/Cards/AttemptDto.cs
namespace WebBFF.Application.Dtos.Cards;

public class AttemptDto
{
    public string UserEmail { get; set; } = string.Empty;
    public DateTime StartedAt { get; set; } = DateTime.UtcNow;
    public DateTime EndedAt { get; set; } = DateTime.UtcNow;
    public bool HasPassed { get; set; } = false;
    public string CardSetId { get; set; } = string.Empty;
}